internal/controller/model_controller.go

```diff
@@ -68,10 +68,23 @@ func (r *ModelReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl
 func (r *ModelReconciler) reconcileModel(ctx context.Context, model *apiv1.Model) (result, error) {
 	log := log.FromContext(ctx)
 
-	if model.Status.Artifacts.URL != "" {
+	if model.Status.Ready {
 		return result{success: true}, nil
 	}
 
+	if model.Status.Artifacts.URL == "" {
+		model.Status.Artifacts.URL = r.Cloud.ObjectArtifactURL(model).String()
+		meta.SetStatusCondition(model.GetConditions(), metav1.Condition{
+			Type:               apiv1.ConditionReconciling,
+			Status:             metav1.ConditionTrue,
+			Reason:             apiv1.ReasonJobComplete,
+			ObservedGeneration: model.Generation,
+		})
+		if err := r.Status().Update(ctx, model); err != nil {
```

Comment:
Remove this status update... This will be too fleeting, not worth the API request.

Reply:
This update request will be sent very soon after this and the URL will be set then.